### ISSUE TITLE
Implement Forest Clean

### ISSUE THREAD

**Issue by LesnyRumcajs** [ChainSafe/forest#1954]
**Issue summary**

Implement a Forest command `clean`. The usage could be along of lines of `forest clean`. The idea is to provide a high-level command to clean-up Forest's data directory, e.g. past snapshots or database that needs to be cleaned because it got too large without playing with dangerous `rm -rf` and guessing where the heck Forest is storing all this data.

It's a bit of a destructive thing to do, so we should prompt the user and explain the severity (e.g. lost database).

**Task summary**

- [ ] implement prompting the user, providing description of the clean-up action to come,
- [ ] implement `--force` to skip prompting the user, e.g. in automatic workflows,
- [ ] implement `--snapshots-only` flag to delete only the snapshots,
- [ ] implement `--database-only` flag to delete only the database,
- [ ] implement reporting how much space was freed thanks to the action. Perhaps it should be included in the first prompt as well.

**Acceptance Criteria**

- [ ] the command should respect the `--chain` flag, e.g. `forest --chain calibnet clean` should not touch anything related to mainnet.
- [ ] user is able to remove old data without looking for Forest data directory and invoking deletes manually, trying to understand Forests directory structure.

**Other information and links**

Feel free to propose your own API providing similar functionality, this is just a suggestion how I roughly see it.

**Comment by hanabi1224** [comment]
Hey @LesnyRumcajs, after looking into this, I have some questions.

Does it make sense to implement different clean func as subcommand of subcommand?  e.g.  `forest fetch clean`.  Also `forest fetch list` might be helpful for someone who cannot remember the default snapshot dir like myself, and then another `forest fetch remove [snapshot]` might be intuitive

**Comment by LesnyRumcajs** [comment]
@hanabi1224 Sounds interesting, though keep in mind it's not only about downloaded snapshots but also the RocksDB.
```
.local/share/forest
❯ ls
calibnet/  filecoin-proof-parameters/  keystore  keystore.json  libp2p/  mainnet/  snapshots/
```

`forest fetch list` sounds a bit bizarre. Might be worth moving some commands around so it's more self-explaining. That is to say, something like `forest snapshot list`. Then maybe the download part should be in `forest snapshot fetch`. Right now it seems to be a mess.

Might be an opportunity for another task that would need to be done before this one? :)

**Comment by hanabi1224** [comment]
I like `forest snapshot ` series commands!

**Comment by hanabi1224** [comment]
Snapshot cleanup will be covered by `snapshot prune` introduced by this PR https://github.com/ChainSafe/forest/pull/2039
